src/api/fileUploadApi.ts

```diff
@@ -0,0 +1,91 @@
+import Axios from 'axios';
+import { getIdToken } from './userApi';
+import { File } from '../types/file.types';
+import { UppyCompletePart } from '../utils/uppy-config';
+
+export enum FileUploadApiPaths {
+  CREATE = '/upload/create',
+  LIST_PARTS = '/upload/listparts',
+  PREPARE = '/upload/prepare',
+  ABORT = '/upload/abort',
+  COMPLETE = '/upload/complete',
+}
+
+export const abortMultipartUpload = async (uploadId: string, key: string) => {
+  const payload = {
+    uploadId,
+    key,
+  };
+
+  const idToken = await getIdToken();
+  const response = await Axios.post(FileUploadApiPaths.ABORT, payload, {
+    headers: {
+      Authorization: `Bearer ${idToken}`,
+    },
+  });
+  return response.data;
+};
+
+export const completeMultipartUpload = async (uploadId: string, key: string, parts: UppyCompletePart[]) => {
+  const payload = {
+    uploadId,
+    key,
+    parts,
+  };
+
+  const idToken = await getIdToken();
+  const response = await Axios.post(FileUploadApiPaths.COMPLETE, payload, {
+    headers: {
+      Authorization: `Bearer ${idToken}`,
+    },
+  });
+  return response.data;
```

Comment:
er tanken å sjekke om det finnes data for så å kaste feil? eller burde det blitt returnert et error-objekt her også? slik at vi sjekker på response.error (som vi har gjort i andre api-kall i andre filer)?

Reply:
Uppy har innebygd feilhåndtering hvis objektet som returneres ikke er riktig. Denne håndterer feil som at "du kan ikke laste opp samme fil to ganger", "ugyldig upload id", og alt mulig annet rart, så jeg tenker egentlig det er best å overlate alt til uppy for nå. Medlingen vil dukke opp som en type snackbar i selve opplastingsfeltet da, og ikke nede i hjørnet som resten av appen